internal/services/azurestackhci/stack_hci_deployment_setting_resource.go

```diff
@@ -0,0 +1,1364 @@
+						ScaleUnits: expandDeploymentSettingScaleUnits(config.ScaleUnit),
+					},
+				},
+			}
+
+			// the resource may exist even validation error
+			metadata.SetID(id)
+
+			// do validation
+			if err := client.CreateOrUpdateThenPoll(ctx, id, payload); err != nil {
+				return fmt.Errorf("validating %s: %+v", id, err)
+			}
+
+			// do deployment
+			payload.Properties.DeploymentMode = deploymentsettings.DeploymentModeDeploy
+			if err := client.CreateOrUpdateThenPoll(ctx, id, payload); err != nil {
+				return fmt.Errorf("deploying %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (StackHCIDeploymentSettingResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.AzureStackHCI.DeploymentSettings
+
+			id, err := deploymentsettings.ParseDeploymentSettingID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			schema := StackHCIDeploymentSettingModel{
+				StackHCIClusterId: deploymentsettings.NewClusterID(id.SubscriptionId, id.ResourceGroupName, id.ClusterName).ID(),
+			}
+
+			if model := resp.Model; model != nil {
+				if props := model.Properties; props != nil {
+					schema.ArcResourceIds = props.ArcNodeResourceIds
+					schema.Version = pointer.From(props.DeploymentConfiguration.Version)
+					schema.ScaleUnit = flattenDeploymentSettingScaleUnits(props.DeploymentConfiguration.ScaleUnits)
+				}
+			}
+
+			return metadata.Encode(&schema)
+		},
+	}
+}
+
+func (StackHCIDeploymentSettingResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 1 * time.Hour,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.AzureStackHCI.DeploymentSettings
+
+			id, err := deploymentsettings.ParseDeploymentSettingID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", id, err)
+			}
+
+			applianceName := fmt.Sprintf("%s-arcbridge", id.ClusterName)
+			applianceId := appliances.NewApplianceID(id.SubscriptionId, id.ResourceGroupName, applianceName)
+
+			log.Printf("[DEBUG] Deleting Arc Resource Bridge Appliance %s generated during deployment", applianceId.ID())
+
+			applianceClient := metadata.Client.ArcResourceBridge.AppliancesClient
+			if err := applianceClient.DeleteThenPoll(ctx, applianceId); err != nil {
+				return fmt.Errorf("deleting %s: %+v", applianceId, err)
```

Comment:
Can we add some contextual information to the wrapped text here, since the user isn't managing this resource directly, it may be ver confusing to have this message presented for a resource they may be unaware of?